### PR TITLE
[ADD] point_of_sale: documentation for pos-based marketing

### PR DESCRIPTION
**Pull request by paan-odoo**
This PR adds the documentation needed to use pos-based marketing.

In this PR, we introduce configuration and flow related to Email and WhatsApp marketing in Point of Sale from storing contact details to sending emails and WhatsApp messages from the point of sale.

task- 4053075

### CODE REVIEW

**Comment by LoredanaLrpz** [comment]
Hi @paan-odoo, thanks so much for your work on this PR! 
I've been reviewing it and noticed a few areas that might need significant changes. To help move things along, I think it might be best if I make some updates directly and push them to the branch. Would you still like me to share my detailed feedback, or should I focus on the revised version instead? Let me know what you think!

**Comment by paan-odoo** [comment]
Hello @LoredanaLrpz 
Yes, I'm fine with that. You can go ahead and push the changes in a separate commit.

Also, Version 18.0 includes additional features that need to be documented. Additionally, there has been a change in the email marketing flow in the master (saas-18.1).

Thanks!

File: content/applications/sales/point_of_sale/pos_based_marketing/email.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
+:show-content:
```
Comment (LoredanaLrpz) [inline content/applications/sales/point_of_sale/pos_based_marketing/email.rst:1]: That is not useful is there's no toctree on the page

File: content/applications/sales/point_of_sale/pos_based_marketing/whatsapp.rst [hunk 2]
```diff
@@ -0,0 +1,45 @@
+:show-content:
```
Comment (LoredanaLrpz) [inline content/applications/sales/point_of_sale/pos_based_marketing/whatsapp.rst:1]: That is not useful is there's no toctree on the page

**Review by LoredanaLrpz — COMMENTED** [review]
Hey, here are the suggestions  I made, I'll still push but that way you know what I started saying =)
Have a nice day

**Comment by paan-odoo** [comment]
Hello @LoredanaLrpz 

I have pushed your suggestions.

Thanks!

**Comment by LoredanaLrpz** [comment]
The commit I pushed just now is **not** the PR's final version ;

**Comment by LoredanaLrpz** [comment]
@robodoo r+

**Pull request closed without merging**